sw/device/silicon_creator/rom_ext/rom_ext_epmp.c

```diff
@@ -6,173 +6,73 @@
 
 #include "sw/device/lib/base/bitfield.h"
 #include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/base/hardened.h"
+#include "sw/device/silicon_creator/lib/epmp_state.h"
 
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
 
-void rom_ext_epmp_unlock_owner_stage_rx(epmp_region_t region) {
-  const int kEntry = 1;
-  epmp_state_configure_tor(kEntry, region, kEpmpPermLockedReadExecute);
-
-  // Update the hardware configuration (CSRs).
-  //
-  // Entry is hardcoded as 1. Make sure to modify hardcoded values if changing
-  // kEntry.
-  //
-  // The `pmp1cfg` configuration is the second field in `pmpcfg0`.
-  //
-  //            32          24          16           8           0
-  //             +-----------+-----------+-----------+-----------+
-  // `pmpcfg0` = |  `pmp3cfg |  `pmp2cfg`| `pmp1cfg` | `pmp0cfg` |
-  //             +-----------+-----------+-----------+-----------+
-
-  CSR_WRITE(CSR_REG_PMPADDR0, ((uint32_t)region.start) >> 2);
-  CSR_WRITE(CSR_REG_PMPADDR1, ((uint32_t)region.end) >> 2);
-  CSR_CLEAR_BITS(CSR_REG_PMPCFG0, 0xff << 8);
-  CSR_SET_BITS(CSR_REG_PMPCFG0, (kEpmpModeTor | kEpmpPermLockedReadExecute)
-                                    << 8);
+#define EPMP_SET(cfg_reg, addr_reg, mask, cfg, addr) \
```

Comment:
The following may be enough to complete hardening for this given that in `rom_ext_epmp_set()` `epmp_state` is  updated separately.

```c
#define EPMP_SET(entry, cfg_reg, addr_reg, mask, cfg, addr) \
  ...
  HARDENED_CHECK_EQ(entry, cfg_reg); \
  ...
```